NDTensors/src/lib/Sectors/src/category_product.jl

```diff
@@ -8,22 +15,33 @@ CategoryProduct(c::CategoryProduct) = _CategoryProduct(categories(c))
 
 categories(s::CategoryProduct) = s.cats
 
-Base.isempty(S::CategoryProduct) = isempty(categories(S))
-Base.length(S::CategoryProduct) = length(categories(S))
-Base.getindex(S::CategoryProduct, args...) = getindex(categories(S), args...)
+const EmptyCategoryProduct = CategoryProduct{Tuple{}}
```

Comment:
```suggestion
const EmptyCategoryProduct{Categories<:Union{Tuple{},NamedTuple{}}} = CategoryProduct{Categories}
```
since both should be interpreted as empty categories, even if we bias towards constructing/converting to the `Tuple{}` version.

Reply:
We can still define:
```julia
EmptyCategory() = CategoryProduct(())
```